conans/model/conan_file.py

```diff
@@ -121,15 +122,18 @@ def __init__(self, display_name=""):
     def serialize(self):
         result = {}
 
-        for a in ("url", "license", "author", "description", "topics", "homepage", "build_policy",
-                  "upload_policy",
-                  "revision_mode", "provides", "deprecated", "win_bash", "win_bash_run"):
+        for a in ("name", "version", "user", "channel", "url", "license",
+                  "author", "description", "topics", "homepage", "build_policy", "upload_policy",
+                  "revision_mode", "provides", "deprecated", "win_bash", "win_bash_run",
+                  "default_options", "options_description"):
             v = getattr(self, a)
             if v is not None:
-                result[a] = v
+                result[a] = str(v)
```

Comment:
default_options and options description are plain dicts, maybe it is not necessary to serialize it to string, but it can be dictionaries?

Reply:
Yep. The issue this solves was `version` being a `Version`. The safer choice would be to special-case it, but this seemed more robust against future changes